agent-shell.el

```diff
@@ -1429,6 +1430,15 @@ Must provide ON-SESSION-INIT (lambda ())."
    :on-success (lambda (response)
                  (map-put! agent-shell--state
                            :session-id (map-elt response 'sessionId))
+                 ;; Store available modes and current mode
+                 (when-let ((modes-data (map-elt response 'modes)))
+                   (setf (map-elt agent-shell--state :available-modes)
```

Comment:
With the term "mode" being so overriden in Emacs, how about we qualify and name something like `:permission-modes`.